src/parser/precedence.js

```diff
@@ -10,6 +10,9 @@ export const PRECEDENCE_DIVIDE = 1;
 export const PRECEDENCE_INCREMENT = 2;
 export const PRECEDENCE_DECREMENT = 2;
 export const PRECEDENCE_ASSIGNMENT = 3;
+export const PRECEDENCE_BITWISE_OR = 7;
+export const PRECEDENCE_BITWISE_XOR = 8;
+export const PRECEDENCE_BITWISE_AND = 9;
```

Comment:
These values likely need to be rebalanced. Bitwise math now will have higher precedence than addition and subtraction, which we don't want. We should more closely resemble JS rules https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence

Reply:
Yep, I agree. I'll try to rebalance the precedences. Regarding the bitwise operator chaining, the issue was fixed by adding the operators to `patch-typecasts.js` file.

Reply:
Thanks. Once you do change the values, write some specs which combine the various arithmetic and bitwise operators to ensure we haven't regressed on anything.

Reply:
So, if I understand the current precedences correctly the new ones should be negative values?

Something like:
```
export const PRECEDENCE_BITWISE_OR = -3;
export const PRECEDENCE_BITWISE_XOR = -2;
export const PRECEDENCE_BITWISE_AND = -1;
```

Reply:
That should work. Having negative values is a bit odd, but it doesn't much matter as long as the semantics are correct.